src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -66,6 +69,29 @@ export class PtyService extends Disposable implements IPtyService {
 		}));
 	}
 
+	private _pendingDetachInstanceRequests: Map<number, (resolved: IProcessDetails | undefined) => void> = new Map();
+	async requestDetachInstance(workspaceId: string, instanceId: number): Promise<IProcessDetails | undefined> {
+		return new Promise<IProcessDetails | undefined>(resolve => {
+			const requestId = ++lastResolvedInstanceRequestId;
+			this._pendingDetachInstanceRequests.set(requestId, resolve);
+			this._onDidRequestDetach.fire({ requestId, workspaceId, instanceId });
+		});
+	}
+
+	async acceptDetachedInstance(requestId: number, persistentProcessId: number): Promise<IProcessDetails | undefined> {
```

Comment:
I think we want `persistentProcessId` to be `number | undefined` here so the renderer can reject the detach request?